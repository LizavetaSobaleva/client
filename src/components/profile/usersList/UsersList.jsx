import React from "react";
import "./usersList.less";
import User from "./user/User";

const UsersList = ({ users, fetchUsers }) => {

  return (
    <>
      <div className="users">
        <div className="users__header">
          <div className="users__name">Name</div>
          <div className="users__email">Email</div>
          <div className="users__status">Status</div>
          <div className="users__size">Used space</div>
        </div>

        <div className="users__list">
          {users && users.length > 0 ? (
            users.map((user) => (
              <User
                key={user._id}
                user={user}
                fetchUsers={fetchUsers}
              />
            ))
          ) : (
            <div>No users found</div>
          )}
        </div>
      </div>
    </>
  );
};

export default UsersList;
