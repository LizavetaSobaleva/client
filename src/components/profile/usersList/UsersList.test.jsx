import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import UsersList from "./UsersList";

jest.mock("./user/User", () => ({ user }) => <div>{user.name}</div>);

const mockStore = configureStore([]);

const users = [
  {
    _id: "1",
    name: "Test User 1",
    email: "test1@example.com",
    usedSpace: 52428800,
    diskSpace: 104857600,
    status: "standard",
  },
  {
    _id: "2",
    name: "Test User 2",
    email: "test2@example.com",
    usedSpace: 209715200,
    diskSpace: 524288000,
    status: "premium",
  },
];

let store;

beforeEach(() => {
  store = mockStore({
    users: users,
  });
  store.dispatch = jest.fn();
});

const renderComponent = () =>
  render(
    <Provider store={store}>
      <UsersList users={users} fetchUsers={jest.fn()} />
    </Provider>
  );

describe("UsersList Component", () => {
  test("renders UsersList without crashing", () => {
    const { asFragment } = renderComponent();
    expect(screen.getByText("Test User 1")).toBeInTheDocument();
    expect(screen.getByText("Test User 2")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays headers correctly", () => {
    renderComponent();
    expect(screen.getByText(/name/i)).toBeInTheDocument();
    expect(screen.getByText(/email/i)).toBeInTheDocument();
    expect(screen.getByText(/status/i)).toBeInTheDocument();
    expect(screen.getByText(/used space/i)).toBeInTheDocument();
  });

  test('displays message when no users are present', () => {
    render(
      <Provider store={store}>
        <UsersList users={[]} fetchUsers={jest.fn()} />
      </Provider>
    );
    
    expect(screen.getByTestId("noUsersMessage")).toBeInTheDocument();
  });
});