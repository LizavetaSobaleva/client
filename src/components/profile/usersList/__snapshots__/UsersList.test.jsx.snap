// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList Component renders UsersList without crashing 1`] = `
<DocumentFragment>
  <div
    class="users"
    data-testid="usersList"
  >
    <div
      class="users__header"
    >
      <div
        class="users__name"
      >
        Name
      </div>
      <div
        class="users__email"
      >
        Email
      </div>
      <div
        class="users__status"
      >
        Status
      </div>
      <div
        class="users__size"
      >
        Used space
      </div>
    </div>
    <div
      class="users__list"
    >
      <div>
        Test User 1
      </div>
      <div>
        Test User 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
